category_encoders/ordinal.py

```diff
@@ -266,8 +266,12 @@ def ordinal_encoding(X_in, mapping=None, cols=None, impute_missing=True, handle_
         else:
             mapping_out = []
             for col in cols:
-                categories = [x for x in pd.unique(X[col].values) if x is not None]
-                categories_dict = {x: i + 1 for i, x in enumerate(categories)}
+                if X[col].dtype.name == 'category':
+                    categories = X[col].cat.categories
+                    categories_dict = {x: i for i, x in enumerate(categories)}
```

Comment:
In the normal case we start the categories at 1, not 0 (so, i+1), to reserve 0 for unknown. We used to use -1 for unknown, but some upstream encoders don't support negative categories, so we changed it recently.